### PR TITLE
Use ui_cancel & ui_accept instead of hardcoded keys in Popups and Dialogs.

### LINKED ISSUE #75431: Popups close when hitting the Esc key, regardless of the input map
### Godot version

v4.0.stable.official [92bee43ad]

### System information

Windows 10

### Issue description

Currently, popups can be closed by using the Esc key. But this completely ignores the game's input map. IIRC popups used to have this closing behavior tied to the `ui_cancel` action, which I think made more sense. I'm guessing it's because of [this line](https://github.com/godotengine/godot/blob/1222d35cdd832b3d4dfd96722d746d312e69a9bb/scene/gui/popup.cpp#L39), as the Esc key is hardcoded there.

### Steps to reproduce

Create a Popup and open it. Even if the input map never references the Esc key, the popup can be closed by pressing the Esc key.

### Minimal reproduction project

N/A

### PR DESCRIPTION
**Pull request by ghost**
Fixed a bug where popups closed using the Escape key, instead of whatever keybind is mapped to "ui_cancel".
Also fixed related issue in Dialog where Escape key was hardcoded instead of respecting "ui_cancel".

- Original Issue: #75431
- [Original Pull Request](https://github.com/godotengine/godot/pull/75518)

_Production edit: Fixes #75431_

### CODE REVIEW

File: scene/gui/dialogs.cpp [hunk 1]
```diff
@@ -39,7 +39,7 @@
 
 void AcceptDialog::_input_from_window(const Ref<InputEvent> &p_event) {
 	Ref<InputEventKey> key = p_event;
-	if (close_on_escape && key.is_valid() && key->is_pressed() && key->get_keycode() == Key::ESCAPE) {
+	if (close_on_escape && key.is_valid() && key->is_pressed() && key->is_action_pressed(SNAME("ui_cancel", false, true))) {
```
Comment (AThousandShips) [inline scene/gui/dialogs.cpp:42]: This is still unnecessary, `is_pressed` isn't needed if you're going to use `is_action_pressed`

Also you need to write `SNAME("ui_cancel")`

**Comment by rypor** [comment]
I also noticed hardcoded keys that can be changed in [file_dialog.cpp](https://github.com/godotengine/godot/blob/master/scene/gui/file_dialog.cpp#L148). Looking at this I would probably change the switch statement to an ifelse block. What do any reviewers thing, and if it should be changed should I add it to this pull request, or start a new issue?

File: scene/gui/dialogs.cpp (hunk 1 above)
Comment (rypor) [reply scene/gui/dialogs.cpp:42]: Got it I missed that is_pressed isn't necessary on the original review.

Also I see I messed up my parenthesis there, fixing it now

**Comment by YuriSizov** [comment]
@rypor You don't need to make a new pull request. Just push or force-push to the same branch, and it will update your submission. I'm going to close the original one, but please keep working on this one instead of making one more if there is a need to make further changes :)

**Comment by AThousandShips** [comment]
You've got a merge commit, to make it up to date you need to do `git rebase master`, and you wiped out your changes to `Tree` by accident

**Comment by rypor** [comment]
> You've got a merge commit, and a bunch of duplicate commits, to make it up to date you need to do `git rebase master`, and you wiped out your changes to `Tree` by accident

Tree uses Input::get_singleton()->is_key_pressed(), which is different from the other two which use InputEvent->is_key_pressed().  
The input is_key_pressed() only accepts keycodes as a parameter, so I am not sure how to apply the mapping. I feel that this could require a bigger change than it seems at first, but if you know of a way to do this please let me know. For now I removed the changes and will later add an issue for Tree

Also thank you, i'll check out the provided link

**Comment by AThousandShips** [comment]
Good, just wasn't sure why you removed that part so wanted to make sure

**Comment by YuriSizov** [comment]
> Tree uses Input::get_singleton()->is_key_pressed()

There is `Input::is_action_pressed()` that you can use. It can be done in this PR as well.

**Pull request closed without merging**